.gitlab-ci.yml

```diff
@@ -0,0 +1,98 @@
+##############################################################################
+# Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and Axom
+# project contributors. See the LICENSE file for details.
+##############################################################################
+
+variables:
+  GIT_SUBMODULE_STRATEGY: recursive
+  PROJECT_ALLOC_NAME: ${CI_PROJECT_NAME}_ci_${CI_PIPELINE_ID}
+  BUILD_ROOT: ${CI_PROJECT_DIR}
+  SLURM_ACCOUNT: -p pdebug
+  LSF_ACCOUNT: -q pdebug
+
+# There are no tests for now
+stages:
+  - l_build
+  - l_build_with_deps
+  - q_allocate_resources
+  - q_build
+  - q_build_with_deps
+  - q_build_devtools
+  - q_release_resources
+
+.pr_workflow:
+  rules:
+    - if: '$CI_COMMIT_BRANCH != "main" && $CI_COMMIT_BRANCH != "develop" && $BUILD_DEPS != "ON"' #run only if ...
+
+.main_workflow:
+  rules:
+    - if: '$CI_COMMIT_BRANCH == "main" || $CI_COMMIT_BRANCH == "develop" || $BUILD_DEPS == "ON"' #run only if ...
+
+####
+# Template
+.srun_build_script:
+  script:
+    #Use pre-existing allocation if any
+    - JOBID=$(squeue -h --name=${PROJECT_ALLOC_NAME} --format=%A)
+    - ASSIGN_ID=$(if [[ -n "${JOBID}" ]]; then echo "--jobid=${JOBID}"; fi)
+    - EXEC_PREFIX="srun ${SLURM_ACCOUNT} ${ASSIGN_ID}"
+   #BUILD + TEST
+    - RESOURCES="-t 10 -N 1"
+    - echo -e "section_start:$(date +%s):build_and_test\r\e[0K
+      Build and test ${CI_PROJECT_NAME}"
+    - ${EXEC_PREFIX} ${RESOURCES} scripts/llnl_scripts/build_src.py -v --host-config ${HOST_CONFIG}
+    - echo -e "section_end:$(date +%s):build_and_test\r\e[0K"
+
+.srun_build_with_deps_script:
+  script:
+    - EXEC_PREFIX="srun ${SLURM_ACCOUNT}"
+    #DEPENDENCIES
+    - RESOURCES="-t 45 -N 1"
+    - echo -e "section_start:$(date +%s):dependencies\r\e[0K
+      Build dependencies and generate host-config file (uberenv)"
+    - ${EXEC_PREFIX} ${RESOURCES} scripts/llnl_scripts/build_tpls.py -v --spec=${SPEC} --directory=${CI_BUILDS_DIR}/axom_tpls/${CI_RUNNER_ID} --mirror=${CI_BUILDS_DIR}/axom_tpls
+    - echo -e "section_end:$(date +%s):dependencies\r\e[0K"
+    #BUILD + TEST
+    - RESOURCES="-t 10 -N 1"
+    - echo -e "section_start:$(date +%s):build_and_test\r\e[0K
```

Comment:
>     - echo -e "section_start:$(date +%s):build_and_test\r\e[0K 

Is there a cleaner way to get colored output?

Reply:
Also, on the gitlab side, I see lines like the following, so perhaps this formulation still needs a bit of tweaking:

```
$ echo -e "section_end:$(date +%s):build_and_test\r\e[0K"
logout
bash: use: command not found
```

Reply:
we can clean this up in a follow up pr